lib/pure/parsecfg.nim

```diff
@@ -622,20 +905,53 @@ proc getSectionValue*(dict: Config, section, key: string, defaultVal = ""): stri
   ## Gets the key value of the specified Section.
   ## Returns the specified default value if the specified key does not exist.
   if dict.hasKey(section):
-    if dict[section].hasKey(key):
-      result = dict[section][key]
+    let kv = dict[section].keyValue
+    if kv.hasKey(key):
+      result = kv[key].value
     else:
       result = defaultVal
   else:
     result = defaultVal
 
 proc setSectionKey*(dict: var Config, section, key, value: string) =
   ## Sets the Key value of the specified Section.
-  var t = newOrderedTable[string, string]()
-  if dict.hasKey(section):
-    t = dict[section]
-  t[key] = value
-  dict[section] = t
+  var tp: SectionItem
+  var kv = <//>OrderedTableRef[string, KeyValueItem]()
+  var kvi: KeyValueItem
+  if dict.hasKey(section): # modify section
+    tp = dict[section]
+    kv = tp.keyValue
+    if kv.hasKey(key): # modify key
+      kvi = kv[key]
+      kvi.value = value
+    else: # add key
+      if key.startsWith("--") or key.startsWith("-"):
+        kvi.keyValueRelated.token = ":"
+      else:
+        kvi.keyValueRelated.token = "="
+      kvi.value = value
+      if not allCharsInSet(value, SymChars): ## Non system character
+        kvi.keyValueRelated.valueStringKind = skString
+      else:
+        kvi.keyValueRelated.valueStringKind = skSymbol
+      kv.add(key, kvi)
+    tp.keyValue = kv
+    dict[section] = tp
+  else: # add section
+    if key.startsWith("--") or key.startsWith("-"):
```

Comment:
On proc that take `char` and if you use a single char string, use the `char` overload, 
because operations over a `char` tends to be more efficient than over a `string`.